### PR TITLE
Delete functions

### PR DESCRIPTION
**Pull request by anitaschiller**
Branch for user stories: Delete member/delete member Entry-card

I added a few functions that make deleting a single member on the "home" page as well as a member entry on the "details" page possible. New is also the DeletionModal component that makes sure that a member can only be deleted upon confirmation.

Next step: Create remember tags

Thank you!! :-)

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -5,18 +5,21 @@ import { v4 as uuid4 } from 'uuid';
...
 import Home from './pages/Home';
 import Navigation from './components/Navigation';
 import { loadFromLocal, saveToLocal } from './lib/localStorage';
 
 function App() {
   const [members, setMembers] = useState(loadFromLocal('members') ?? []);
-  const orderedMembers = members.slice().sort(compareFirstName);
+  const [isShown, setIsShown] = useState(false);
+  const [remainingMembers, setRemainingMembers] = useState(members);
+  console.log('remainingMebers', remainingMembers);
```
Comment (basti-n) [inline client/src/App.js]: No console.logs on main.

File: client/src/pages/Details.js [hunk 2]
```diff
@@ -30,12 +35,15 @@ export default function Details({ member, updateMember, members }) {
...
+      //Add valid entry to entries array and set entry to initial state afterwards
       const newEntry = { ...entry, id: uuid4() };
       const memberEntries = [...entries, newEntry];
       setEntries(memberEntries);
       setEntry(initialEntry);
 
-      const updatedMember = { ...member, entries: memberEntries };
+      //Update the newMember with the new entries and send the updatedMember to App.js
+      const updatedMember = { ...newMember, entries: memberEntries };
+      setNewMember(updatedMember);
```
Comment (basti-n) [inline client/src/pages/Details.js:43]: Both these functions seem to always be called together. Maybe useful to encapsulate the calling in a separate function. This new function would then make sure both functions are called.

File: client/src/App.js [hunk 3]
```diff
@@ -26,30 +29,38 @@ function App() {
...
-
   function updateMember(updatedMember) {
     const upToDateMembers = members.filter(
       (member) => member.id !== updatedMember.id
     );
     setMembers([...upToDateMembers, updatedMember]);
   }
 
+  function openModal(idToDelete) {
+    console.log('deleteMember');
```
Comment (miriamhaenle) [inline client/src/App.js]:
```suggestion

```

File: client/src/App.js [hunk 4]
```diff
@@ -26,30 +29,38 @@ function App() {
...
+    setIsShown(true);
+  }
+
+  function denyDeletion() {
+    setMembers(members);
+    setIsShown(false);
+  }
+
+  function confirmDeletion() {
+    console.log('confirmDeletion');
```
Comment (miriamhaenle) [inline client/src/App.js]:
```suggestion
```

File: client/src/pages/Details.js [hunk 5]
```diff
@@ -13,13 +13,18 @@ export default function Details({ member, updateMember, members }) {
     remember: '',
   };
 
+  //Takes current state of the member out of the members Array.
+  const [newMember, setNewMember] = useState(
+    members.find((newMember) => newMember.id === member.id)
+  );
+  console.log('newMember', newMember);
```
Comment (miriamhaenle) [inline client/src/pages/Details.js]:
```suggestion
```

File: client/src/pages/Details.js [hunk 6]
```diff
@@ -13,13 +13,18 @@ export default function Details({ member, updateMember, members }) {
     remember: '',
   };
 
+  //Takes current state of the member out of the members Array.
+  const [newMember, setNewMember] = useState(
+    members.find((newMember) => newMember.id === member.id)
```
Comment (miriamhaenle) [inline client/src/pages/Details.js:17]: Maybe you can encapsulate this into its own function with a descriptive name and thus won't need the comment in line 16
`const findCurrentMember = (newMember) => members.find(member => member.Id === newMember.id)`

**Review by miriamhaenle — CHANGES_REQUESTED** [review]
Hi Anita, 
nice job again! I added a few comments. 
Please also take a look  again at your files. There are a lot of console.logs still present. It is not considered good practice to merge console.logs into main (only use it while developing). I did highlight a few but please check again and remove them. 

I also noticed quite a few comments, especially in your Details.js. Please also take a look again if these comments are really needed in order to understand your code. Ideally comments are use scarcely and only where the code itself can not convey its intention fully.

On a more general note you can take a look again at your colors. You have defined some global variables, sometimes use HTML color names and sometimes hex values. Try to stick with one approach if possible :) Something you could change later on in the project.

👍

**Comment by anitaschiller** [comment]
> Hi Anita,
> nice job again! I added a few comments.
> Please also take a look again at your files. There are a lot of console.logs still present. It is not considered good practice to merge console.logs into main (only use it while developing). I did highlight a few but please check again and remove them.
> 
> I also noticed quite a few comments, especially in your Details.js. Please also take a look again if these comments are really needed in order to understand your code. Ideally comments are use scarcely and only where the code itself can not convey its intention fully.
> 
> On a more general note you can take a look again at your colors. You have defined some global variables, sometimes use HTML color names and sometimes hex values. Try to stick with one approach if possible :) Something you could change later on in the project.
> 
> 👍

Ouh yes, I totally forgot to clean up this time, but will do before I merge! Thank you for looking through and all your comments! :-)

**Pull request merged**